shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-core/src/main/java/org/apache/shardingsphere/data/pipeline/core/spi/check/consistency/DataMatchSingleTableDataCalculator.java

```diff
@@ -124,5 +125,39 @@ public void init() {
         private final int recordCount;
         
         private final Collection<Collection<Object>> records;
```

Comment:
Could we extract `Collection<Object>` into another class that includes special `equals` and `hashCode` impl to simplify `CalculatedResult`, and also add unit test for this new class.

Reply:
Refactor later